src/core/String.cpp

```diff
@@ -73,8 +73,7 @@ namespace String
         for (size_t i = 0; i < minSize; i++)
         {
             *dst++ = *src;
-            if (*src == '\0') break;
-            *src++;
+            if (*src++ == '\0') break;
```

Comment:
It's less lines of code. (and the other way was giving a warning due to the dereference). If you want, I can easily revert that.